app/models/invoice.rb

```diff
@@ -14,6 +14,14 @@ def self.revenue_by_date(date)
       .sum("invoice_items.unit_price * invoice_items.quantity")
   end
 
+  def self.single_merchant_revenue_by_date(date, merchant_id)
```

Comment:
Looks great! It's so similar to revenue ... I wonder if we can set merchant_id to a default value that returns all merchants and kill both birds with one stone? Just a thought